kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -214,23 +215,14 @@ translatePredicate translateUninterpreted predicate =
             { applicationSymbolOrAlias
             , applicationChildren
             }
-      = case
-            getSmtlib (symAttributes smtlibTools applicationSymbolOrAlias)
-            <|>
-            getSmthook (symAttributes smthookTools applicationSymbolOrAlias)
-        of
+      = case translateSymbol applicationSymbolOrAlias of
```

Comment:
Would it be better to use `translateName` here because the sorts are not needed? That would eliminate the need for `applicationSorts` in `MetadataTools`: the sorts are only used by `declareSymbol`, which already has the full symbol declaration sentence. I noticed a lot of test changes because of that addition which are otherwise not related to the SMT changes.

Reply:
No. I added a note in the `translateSymbol` documentation to explain why:

```
Note that it's not enough to call `translateName` to check whether a symbol
was/can be declared since it only checks that the symbol itself is declarable,
but it does not check that the symbol's sort is declarable.
```

Reply:
> check whether a symbol was/can be declared

Could `declareSymbols` update `MetadataTools` with *that* information ("symbol was/was not declared") instead? I'm satisfied with the not, but just trying to think of ways to modularize the change. You can consider that, and I will approve the pull request.

Reply:
I've talked with Virgil and he agreed to look into this when he gets back on Monday.  Hence, I will merge this as-is and we'll move the card into Cleanup.